lib/ansible/modules/notification/say.py

```diff
@@ -68,18 +68,17 @@ def main():
 
     msg = module.params['msg']
     voice = module.params['voice']
-
-    executable = module.get_bin_path('say')
-    if not executable:
-        executable = module.get_bin_path('espeak')
-    elif get_platform() != 'Darwin':
-        # 'say' binary available, it might be GNUstep tool which doesn't support 'voice' parameter
-        voice = None
-        module.warn("'say' executable found but system is '%s': ignoring voice parameter" % get_platform())
-
-    if not executable:
-        module.fail_json(msg="Unable to find either 'say' or 'espeak' executable")
-
+    possibles = ('say', 'espeak', 'espeak-ng')
+
+    for possible in possibles:
+        executable = module.get_bin_path(possible)
+        if executable and get_platform() != 'Darwin':
+            # 'say' binary available, it might be GNUstep tool which doesn't support 'voice' parameter
+            voice = None
+        elif executable:
+            break
+        else:
```

Comment:
not sure if you understood the for/else construct, this would fail if it didnt find the executable at first shot? not sure i understand the if either